stable_baselines/common/vec_env/vec_normalize.py

```diff
@@ -88,48 +89,76 @@ def step_wait(self):
         where 'news' is a boolean vector indicating whether each element is new.
         """
         obs, rews, news, infos = self.venv.step_wait()
-        self.ret = self.ret * self.gamma + rews
         self.old_obs = obs
-        obs = self._normalize_observation(obs)
+        self.old_rews = rews
+
+        if self.norm_obs:
+            if self.training:
+                self.obs_rms.update(obs)
+            obs = self.normalize_obs(obs)
+
         if self.norm_reward:
             if self.training:
-                self.ret_rms.update(self.ret)
-            rews = np.clip(rews / np.sqrt(self.ret_rms.var + self.epsilon), -self.clip_reward, self.clip_reward)
+                self._update_reward(rews)
+            rews = self.normalize_reward(rews)
         self.ret[news] = 0
         return obs, rews, news, infos
 
-    def _normalize_observation(self, obs):
+    def _update_reward(self, rews):
+        self.ret = self.ret * self.gamma + rews
+        self.ret_rms.update(self.ret)
+
+    def normalize_obs(self, obs):
         """
-        :param obs: (numpy tensor)
+        Normalize observations using this VecNormalize's observations statistics.
+        Calling this method does not update statistics.
+
+        :param obs: (numpy float)
```

Comment:
can we make these type annotations? We support this now in SB.

Reply:
done, I can also annotate the rest of the file if that's helpful